Support column_list in CONTAINS clause

jaspal007 opened the pull request:
### Description

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish.

### Issue
Babelfish currently lacks support for CONTAINS clause queries with multiple columns. The `sys.replace_special_chars_fts` function at the TDS endpoint only accepts a single column argument of type text, resulting in syntax errors when attempting multi-column queries.

### Solution
1. Enhanced syntax recognition:
   - Added new rules in `gram-tsql-rule.y` to properly identify CONTAINS clause syntax
   - Modified function declarations to accept column lists as arguments

2. Function improvements:
   - Updated `sys.replace_special_chars_fts` to handle variable number of columns
   - Modified `TsqlExpressionContains` to process column lists uniformly
   - Standardized input handling: single or multiple columns are now processed as lists

These modifications enable Babelfish to correctly process CONTAINS clause queries across multiple columns while maintaining backward compatibility.

Task: BABEL-4239
Authored-by: Jaspal Singh [ijaspals@amazon.com]
Signed-off-by: Jaspal Singh [ijaspals@amazon.com]

### Issue resolved
BABEL-4239

### Test Scenarios Covered ###
* **Use case based -** **YES**


* **Boundary conditions -** **YES**


* **Arbitrary inputs -**


* **Negative test cases -** **YES**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

roshan0708 commented:
Please resolve all build actions.

roshan0708 commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -2777,21 +2777,73 @@ type_name(PG_FUNCTION_ARGS)
+
+
+		/* Modify the input_str in place */
+		if(input_str != NULL)
+		{
+			col_str = replace_special_chars_fts_impl(input_str);
+		}
+
+		if (col_str != NULL) 
+		{
+			/* Append new columns to the output_str  */
+			appendStringInfoString(&buf, col_str);
```
`col_str` may be uninitialized if `input_str` is `NULL`, which isn't safe. Initializing it to `NULL` ensures fail-safe behavior. Make sure to check for other such variable initialization as well, as at some point, it could lead to storing garbage value.

jaspal007 replied:
```col_str``` is initialized to ```NULL``` to ensure fail-safe check and prevent any garbage value.

roshan0708 commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -2777,21 +2777,73 @@ type_name(PG_FUNCTION_ARGS)
+	/* Deconstruct the array to extract elements and nulls from the array*/
+	deconstruct_array(array, TEXTOID, -1, false, 'i', &elements, &nulls, &nelems);
+
+	for(int i = 0; i < nelems; i++)
```
Nit: code formatting and indentation.

jaspal007 replied:
Resolved the indentation issue

roshan0708 commented on contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--5.0.0--5.1.0.sql:
```diff
@@ -112,6 +112,13 @@ CREATE OR REPLACE FUNCTION sys.hashbytes(IN alg sys.VARCHAR, IN data sys.bbf_var
 AS 'babelfishpg_tsql', 'hashbytes' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
 GRANT EXECUTE ON FUNCTION sys.hashbytes(IN alg sys.VARCHAR, IN sys.bbf_varbinary) TO PUBLIC;
 
+-- This function performs replacing special characters to their corresponding unique hashes
+-- in the search condition or the full text search CONTAINS predicate
+CREATE OR REPLACE FUNCTION sys.replace_special_chars_fts(VARIADIC texts text[]) RETURNS TEXT AS 
+'babelfishpg_tsql', 'replace_special_chars_fts'
+LANGUAGE C IMMUTABLE STRICT;
+GRANT EXECUTE ON FUNCTION sys.replace_special_chars_fts TO PUBLIC;
+
```
This function already exists in previous versions, which will cause upgrade failures due to non-unique functions. We need to deprecate and drop the older function before adding this new one.

jaspal007 replied:
The previous version of this function has been deprecated and recreated to serve as a entry point for previous version and now deprecated C function that processes a single column as ```text```.

roshan0708 commented on contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c:
```diff
@@ -782,7 +784,7 @@ is_json_query(List *name)
 * where pgconfig = babelfish_fts_contains_pgconfig('<contains_search_condition>')
```
Nit: Update the function description with `column_list` support as well.

jaspal007 replied:
Updated the function description with support for ```column_list```

roshan0708 commented on contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c:
```diff
@@ -802,73 +804,88 @@ TsqlExpressionContains(char *colId, Node *search_expr, core_yyscan_t yyscanner)
+		char 		*dot1, *dot2;
+
+		/* Find the first and second dots in the column identifier */
+		dot1 = strchr(colName, '.');
```
`strchr` function is used on `colName`, but if `colName` is `NULL`, it could lead to a null pointer dereference. Better to check this before assigning value to `dot1`.

jaspal007 replied:
Added the `NULL-safety` check before assigning colName to `strchr()` to avoid any potential `NULL` pointer dereference.

roshan0708 commented on contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c:
```diff
@@ -802,73 +804,88 @@ TsqlExpressionContains(char *colId, Node *search_expr, core_yyscan_t yyscanner)
+
+	for(int i = 0; i < len; i++)
+	{
+		Node 		*col;
```
Better to initialize `col` as `NULL` for safety.

jaspal007 replied:
Initialized `col` to `NULL` for safety.

roshan0708 commented on contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c:
```diff
@@ -802,73 +804,88 @@ TsqlExpressionContains(char *colId, Node *search_expr, core_yyscan_t yyscanner)
+	}
+	
+    /* Create a function call for replace_special_chars_fts(column_list) */
     replaceSpecialCharsFunc = (Node *) makeFuncCall(TsqlSystemFuncName("replace_special_chars_fts"), replaceSpecialCharsArgs, COERCE_EXPLICIT_CALL, -1);
```
To avoid memory leaks, ensure that `replaceSpecialCharsArgs` is freed when it's no longer needed. Ditto for other such variables which could have potential memory leak issue.

jaspal007 replied:
Freeing `replaceSpecialCharsArgs` with `pfree()` is going to cause a dangling pointer error as we are assigning its reference to `replaceSpecialCharsFunc` of the `sys.replace_special_chars_fts()` that is going to be passed into `args` to return `to_tsvector`.

roshan0708 commented on test/JDBC/expected/fts-contains-multicol-vu-verify.out:
```diff
@@ -0,0 +1,1339 @@
+
+
+-- negative test
+-- should throw an error as the columns txt3 and txt4 are not fulltext indexed, BUT DOES NOT
```
Better to remove this since this is not an expected behavior and is a bug, for negative tests, we need test cases to handle - non existent column or wrong column name, null values, empty search string,

roshan0708 commented on test/JDBC/expected/fts-contains-multicol-vu-verify.out:
```diff
@@ -0,0 +1,1339 @@
+
+
+-- negative test
+-- should throw an error as chr is not fulltext indexed, BUT DOES NOT
+SELECT * FROM fts_schema.test_tb1 WHERE contains((val, chr), 'four')
```
ditto.

roshan0708 commented on test/JDBC/input/full_text_search/fts-contains-multicol-vu-cleanup.sql:
```diff
@@ -0,0 +1,26 @@
+-- tsql user=jdbc_user password=12345678
```
This is redundant since your test file is in `.sql` format. You can remove this from all the test files.

jaspal007 replied:
Resolved the issue.

roshan0708 commented:
Let’s add more tests for the following scenarios:

- Incorrect dot operator naming convention in column names.
- Large number of columns in the `CONTAINS()` clause.
- Case where there is keyword or phrase overlapping in different columns.
- All syntax error failures, like empty or null column/search strings.
- Test combinations of string data types like `varchar`, `char`, `text`, `nvarchar`, `nchar`, and `ntext`.
- Consider combinations of other data types as well.

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Added a few more test cases

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Added a few more test cases

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Added a few more test cases

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Added a few more test cases

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
jdbc db collation

Commit:
parallel safe

Commit:
parallel safe

roshan0708 commented on test/JDBC/jdbc_schedule:
```diff
@@ -8,8 +8,21 @@
 all
 
+
+# JDBC Fulltext search tests
+fts-contains-multicol-vu-prepare
+fts-contains-multicol-vu-verify
+fts-contains-multicol-vu-cleanup
+fts-contains-utils-vu-prepare
+fts-contains-utils-vu-verify
+fts-contains-utils-vu-cleanup
+fts-contains-vu-prepare
+fts-contains-vu-verify
+fts-contains-vu-cleanup
```
This change is not required, please revert this. `all` flag covers all the test cases, so we don't need to explicitly mention new test files.

jaspal007 replied:
The test cases are added in schedule files and `jdbc_schedule` is set to test `all` files

Commit:
version upgrade

Commit:
utils file

Commit:
view test

Commit:
before files

Commit:
before files

Commit:
ignore before files

Commit:
re commit

Commit:
different function

Commit:
different function

jaspal007 committed:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. To check the version upgrade failure, recreated the deprecated functions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

jaspal007 committed:
Merge branch 'babelfish-for-postgresql:BABEL_5_X_DEV' into BABEL-4239-5_X

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. To check version upgrades failures, recreated deprecated sql functions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

roshan0708 commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -2794,6 +2799,77 @@ replace_special_chars_fts(PG_FUNCTION_ARGS)
+	initStringInfo(&buf);
+
+	/* Deconstruct the array to extract elements and nulls from the array */
+	deconstruct_array(array, TEXTOID, -1, false, 'i', &elements, &nulls, &nelems);
```
Use `TYPALIGN_INT` instead of `'i'`.

roshan0708 commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -2794,6 +2799,77 @@ replace_special_chars_fts(PG_FUNCTION_ARGS)
+			if(i<nelems-1){
+				appendStringInfo(&buf, "||' '||");
+			}
+			pfree(col_str);
```
Nit: code formatting

roshan0708 commented on test/JDBC/jdbc_schedule:
```diff
@@ -8,8 +8,25 @@
+ignore#!#fts-contains-utils_before_17_2-vu-cleanup
+#fts-contains-vu-prepare
+#fts-contains-vu-verify
+#fts-contains-vu-cleanup
```
Ignore the before test files only, rest is redundant, please remove this.

roshan0708 commented on test/JDBC/jdbc_schedule:
```diff
@@ -8,8 +8,25 @@
 # 6. If you want the framework to not run certain files, use: ignore#!#<test name>
 
+
+
```
Irrelevant change, please revert this.

roshan0708 commented on test/JDBC/upgrade/16_2/schedule:
```diff
@@ -228,6 +228,7 @@ fulltextserviceproperty
 FULLTEXT_INDEX
 fts-contains
 fts-contains-utils
+#fts-contains-utils_before_17_2
```
These are not running in 16_2 -> latest upgrade right? In that case, you can completely remove this line. Check for other schedule files as well.

roshan0708 replied:
Also, can you point where these before tests are being run? I can't see the `schedule` file change for this.

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

jaspal007 committed:
Merge branch 'babelfish-for-postgresql:BABEL_5_X_DEV' into BABEL-4239-5_X

jaspal007 committed:
Merge branch 'babelfish-for-postgresql:BABEL_5_X_DEV' into BABEL-4239-5_X

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Merge remote-tracking branch 'refs/remotes/origin/BABEL-4239-5_X' into BABEL-4239-5_X

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. Some utility functions are deprecated to create new function definitions. To check the version upgrade failure due to deprecation, the deprecated functions are recreated to provide an entry point to the queries to extend the support in newer versions.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

jaspal007 committed:
Merge branch 'BABEL_5_X_DEV' into BABEL-4239-5_X

Commit:
changes

Commit:
message

Commit:
message

jaspal007 committed:
Merge branch 'babelfish-for-postgresql:BABEL_5_X_DEV' into BABEL-4239-5_X

Commit:
ft_indexed column

Commit:
message

Commit:
message

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. The support for multiple column fulltext search over CONTAINS clause is extended with validation check over columns if they are a part of existing fulltext index. The fulltext index can be created over 32 columns for a single table.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. The support for multiple column fulltext search over CONTAINS clause is extended with validation check over columns if they are a part of existing fulltext index. The fulltext index can be created over 32 columns for a single table.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. The support for multiple column fulltext search over CONTAINS clause is extended with validation check over columns if they are a part of existing fulltext index. The fulltext index can be created over 32 columns for a single table.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Support column_list in CONTAINS clause

Previously in Babelfish, the feature of fulltext search for simple terms in CONTAINS clause was not supported for multiple columns. The TSQL syntax of Fulltext search for simple terms in CONTAINS clause over multiple columns was unidentifiable over TDS endpoint of Babelfish and would throw syntax error. To fix this, we have created a new grammar to support the TSQL multiple column Fulltext search syntax for simple terms in CONTAINS clause over TDS endpoint of Bablefish. The support for multiple column fulltext search over CONTAINS clause is extended with validation check over columns if they are a part of existing fulltext index. The fulltext index can be created over 32 columns for a single table.

TASK: BABEL-4239
Signed-off-by: Jaspal Singh ijaspals@amazon.com

Commit:
Revert "Support column_list in CONTAINS clause"

This reverts commit 9c9e5cedab3bb72a2edf0714e9cbd2948d705140.

Commit:
Revert "Support column_list in CONTAINS clause"

This reverts commit 9c9e5cedab3bb72a2edf0714e9cbd2948d705140.

Commit:
memessage

jaspal007 committed:
Merge branch 'BABEL_5_X_DEV' into BABEL-4239-5_X

roshan0708 committed:
Support syntax option INLINE = {ON | OFF} in CREATE FUNCTION (#3555)

Currently, Babelfish throws syntax error when INLINE option is used in CREATE FUNCTION statements. The issue arises
from ANTLR's inability to recognise this token due to missing grammar in CREATE FUNCTION. To resolve this, we added
the INLINE = (ON | OFF) grammar to ANTLR and implemented the logic to ignore it during parsing time.

Task: BABEL-3719
Signed-off-by: Roshan Kanwar <rskanwar@amazon.com>

ahmed-shameem committed:
Enable MVU (16.latest to 17.latest) for LIKE to ILIKE transformation (#3575)

Commit:
check version

Commit:
before files

Commit:
verify

jaspal007 committed:
Merge branch 'babelfish-for-postgresql:BABEL_5_X_DEV' into BABEL-4239-5_X
